src/index.js

```diff
@@ -275,10 +269,10 @@ class UglifyJsPlugin {
           // Handling warnings
           if (warnings && warnings.length > 0) {
             warnings.forEach((warning) => {
-              const builtWarning = this.buildWarning(
+              const builtWarning = UglifyJsPlugin.buildWarning(
                 warning,
                 file,
-                inputSourceMap,
+                sourceMap,
```

Comment:
Also `sourceMap` can be undefined here, because warning !== error (catch block)

Reply:
/cc @lojjic yep, my mistake, but if no error(s) - no source map

Reply:
again mistake, sorry, very tired

Reply:
I'm not following you. `sourceMap` can be `null`, if there is no valid `inputSourceMap`, but that's expected AFAICT.

Reply:
Ah, ok! No worries.